src/results.jl

```diff
@@ -80,14 +80,14 @@ function show_result(io::IO, rsp::TransactionResponse)
     rsp.metadata === nothing && return
     rsp.results === nothing && return
 
-    for index in eachindex(rsp.metadata)
-        println(io, rsp.metadata[index]["relationId"])
-        tuples = zip(rsp.results[index][2]...)
+    for (idx, relation_metadata) in enumerate(rsp.metadata.relations)
+        show_relation_id(io, relation_metadata.relation_id)
+        tuples = zip(rsp.results[idx][2]...)
         # Reuse julia's array printing function to print this array of tuples.
         Base.print_array(io, collect(tuples))
 
         # Print trailing newline
-        if index !== last(eachindex(rsp.metadata))
+        if idx !== length(rsp.metadata.relations)
```

Comment:
```suggestion
        if idx !== lastindex(rsp.metadata.relations)
```
After reading Yuri's blog post (https://yuri.is/not-julia/), I think the ask from the julia community is to try to always compare iterators using the "index" API, rather than directly checking against an array's length. The idea is that this code will be resilient to using slices and other weird arrays that aren't indexed `1:N`. I'm not sure if it really matters for this code, which is not generic, but it's maybe nicer to read anyway, and seems like a good habit to get into? 😊

Reply:
Love it :)